src/mito2/src/read/last_row.rs

```diff
@@ -60,6 +67,151 @@ impl BatchReader for LastRowReader {
     }
 }
 
+/// Cached last row reader for specific row group.
+/// If the last rows for current row group are already cached, this reader returns the cached value.
+/// If cache misses, [RowGroupLastRowReader] reads last rows from row group and updates the cache
+/// upon finish.
+pub(crate) enum RowGroupLastRowCachedReader {
+    /// Cache hit, reads last rows from cached value.
+    Hit(LastRowCacheReader),
+    /// Cache miss, reads from row group reader and update cache.
+    Miss(RowGroupLastRowReader),
+}
+
+impl RowGroupLastRowCachedReader {
+    pub(crate) fn new(
+        file_id: FileId,
+        row_group_idx: usize,
+        cache_manager: Option<CacheManagerRef>,
+        row_group_reader: RowGroupReader,
+    ) -> Self {
+        let key = SelectorResultKey {
+            file_id,
+            row_group_idx,
+            selector: TimeSeriesRowSelector::LastRow,
+        };
+
+        let Some(cache_manager) = cache_manager else {
+            return Self::Miss(RowGroupLastRowReader::new(key, row_group_reader, None));
+        };
+        if let Some(value) = cache_manager.get_selector_result(&key) {
+            let schema_matches = value.projection
+                == row_group_reader
+                    .context()
+                    .read_format()
+                    .projection_indices();
+            if schema_matches {
+                // Schema matches, use cache batches.
+                Self::Hit(LastRowCacheReader { value, idx: 0 })
+            } else {
+                Self::Miss(RowGroupLastRowReader::new(
+                    key,
+                    row_group_reader,
+                    Some(cache_manager),
+                ))
+            }
```

Comment:
The cache hit metric is incorrect, we may pass the projection indices to the manager.